src/test/datascience/notebook/kernelSelection.vscode.test.ts

```diff
@@ -0,0 +1,256 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { assert } from 'chai';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as sinon from 'sinon';
+import { commands, Uri } from 'vscode';
+import { CellDisplayOutput } from '../../../../types/vscode-proposed';
+import { IPythonExtensionChecker } from '../../../client/api/types';
+import { IVSCodeNotebook } from '../../../client/common/application/types';
+import { BufferDecoder } from '../../../client/common/process/decoder';
+import { ProcessService } from '../../../client/common/process/proc';
+import { IDisposable } from '../../../client/common/types';
+import { INotebookEditorProvider } from '../../../client/datascience/types';
+import { IInterpreterService } from '../../../client/interpreter/contracts';
+import { getOSType, IExtensionTestApi, OSType, waitForCondition } from '../../common';
+import { EXTENSION_ROOT_DIR_FOR_TESTS, IS_NON_RAW_NATIVE_TEST, IS_REMOTE_NATIVE_TEST } from '../../constants';
+import { closeActiveWindows, initialize, IS_CI_SERVER } from '../../initialize';
+import { openNotebook } from '../helpers';
+import {
+    assertHasTextOutputInVSCode,
+    canRunNotebookTests,
+    closeNotebooksAndCleanUpAfterTests,
+    createTemporaryNotebook,
+    deleteAllCellsAndWait,
+    executeActiveDocument,
+    insertCodeCell,
+    startJupyterServer,
+    trustAllNotebooks,
+    waitForExecutionCompletedSuccessfully,
+    waitForKernelToChange,
+    waitForKernelToGetAutoSelected
+} from './helper';
+
+/* eslint-disable no-invalid-this, , , @typescript-eslint/no-explicit-any */
+suite('DataScience - VSCode Notebook - Kernel Selection', function () {
```

Comment:
Tests for switching kernels & ensuring we use the right interpreter
Covers all of remote, raw & Jupyter scenarios